R/coerce.R

```diff
@@ -7,15 +7,26 @@ coerce <- function(x, type) {
 coerce_lgl <- function(x) coerce(x, "logical")
 coerce_int <- function(x) coerce(x, "integer")
 coerce_dbl <- function(x) coerce(x, "double")
-coerce_chr <- function(x) coerce(x, "character")
+coerce_chr <- function(x) {
+  local_deprecation_user_env()
+  coerce(x, "character")
+}
 
 
-deprecate_to_char <- function(type, call = caller_env()) {
+deprecate_to_char <- function(type) {
   lifecycle::deprecate_warn(
     "1.0.0",
     I(paste0("Automatic coercion from ", type, " to character")),
-    I("an explicit call to as.character() within map_chr()"),
+    I("an explicit call to `as.character()` within `map_chr()`"),
     always = TRUE,
-    user_env = call
+    user_env = the$deprecation_user_env
+  )
+}
+
+local_deprecation_user_env <- function(frame = caller_env()) {
```

Comment:
The user env is the caller of the `map_` function. You can take the grand-parent frame here or pass it through by argument:

```suggestion
local_deprecation_user_env <- function(user_env = caller_env(2), frame = caller_env()) {
```